### PR TITLE
fix integration tests

### LINKED ISSUE #709: Playwright in CI is broken
The [last deployment to staging](https://github.com/sillsdev/languageforge-lexbox/actions/runs/8612853179) uncovered a few problems in our Playwright CI configuration:

**High Priority**, because it's preventing us from evaluating/getting results from our Playwright tests.

From [this job](https://github.com/sillsdev/languageforge-lexbox/actions/runs/8612853179/job/23603097956):
- [x] The steps "Password protect Playwright traces" and "Upload Playwright results" are both configured to run even though we don't want to run Playwright tests in that job
- [x] "Upload Playwright results" runs even if "Password protect Playwright traces" fails, which is a bit of a security concern (and was likely not introduced by our transition to Node)

From [this job](https://github.com/sillsdev/languageforge-lexbox/actions/runs/8612853179/job/23603098262):
- [x] Playwright tests fail to run with the error: `line 1: pnpm: command not found`
- [x] "Integration tests (.NET)" did not run, because "Playwright setup" failed. The .NET tests should run regardless of whether Playwright has issues.

### PR DESCRIPTION
**Pull request by hahn-kev**
should close #709

### CODE REVIEW

File: .github/workflows/integration-test.yaml
Comment (rmunn) [inline .github/workflows/integration-test.yaml:69]: The https://github.com/marketplace/actions/setup-pnpm docs show running setup-node before running the pnpm setup action. However, the setup-node docs say "package manager should be preinstalled" if you're using the `cache` input, so this should be okay if node comes preinstalled on the self-hosted runner.

**Review by rmunn — APPROVED** [review]
LGTM. I started working on this before seeing your PR (I was on a different issue), but didn't spend too much time on my version and yours is more complete. So let's merge this and I'll not even create a PR from my version.

**Comment by rmunn** [comment]
While we're in here, I suggest changing the job name on line 47 from `... on ${{ vars.TEST_SERVER_HOSTNAME }}` to `... on ${{ inputs.environment }}`, because `vars.TEST_SERVER_HOSTNAME` is apparently inaccessible at the time the job name is being calculated and is returning an empty string.

**Comment by rmunn** [comment]
Tests are now running, and all problems from https://github.com/sillsdev/languageforge-lexbox/issues/709 are now solved. Marking as ready for review.

File: .github/workflows/integration-test.yaml
Comment (rmunn) [reply .github/workflows/integration-test.yaml:75]: Solved: we had to run setup-node twice. Once before setting up pnpm (without the `cache` input param), and once afterwards with the `cache: pnpm` setting.

**Comment by rmunn** [comment]
@myieye - Looks like I should ask you for a review in addition to my earlier one, since both Kevin and I have worked on this.

File: frontend/playwright.config.ts
Comment (myieye) [inline frontend/playwright.config.ts:57]: I really prefer `on` as my comment on the line above (somewhat poorly) explains.
I've been in too many situations where I wanted to see what a successful trace looked like as part of debugging an unsuccessful execution. So, I like having successful traces available.

**Review by hahn-kev — COMMENTED** [review]
nice job Robin! Looking forward to getting this merged in

File: .github/workflows/integration-test.yaml
Comment (hahn-kev) [inline .github/workflows/integration-test.yaml:125]: I don't think we need a 1 minute timeout anymore, the tests are running much faster, I think if we have to set the global timeout to 1 minute then we need to solve that some other way, that's just too long to wait.

File: frontend/playwright.config.ts
Comment (rmunn) [reply frontend/playwright.config.ts:57]: I'll try setting this back to `on`, but it might result in no traces at all because the `7z` process runs out of memory and gets killed again. The `trace.zip` files are quite large. However, my setting 7z to use no compression may have fixed that problem; we'll see.

File: .github/workflows/integration-test.yaml
Comment (rmunn) [reply .github/workflows/integration-test.yaml]: Done in commit 9c3718eb.

**Comment by rmunn** [comment]
@hahn-kev - Want to throw an approval on this before you go to bed, so I can get it merged in today? I'll make sure Tim's okay with it too, of course.

File: frontend/playwright.config.ts
Comment (rmunn) [reply frontend/playwright.config.ts]: Done in commit 014af79e7b9fec6f6cab37cdf74e289ffa18b205, and the 7z process did not get OOM killed. Looks like this is working.

**Pull request merged**
